src/libs/actions/PersonalDetails.js

```diff
@@ -236,6 +237,53 @@ function setPersonalDetails(details) {
     mergeLocalPersonalDetails(details);
 }
 
+/**
+ * Fetches the Currency preferences based on location
+ * @param {bool} withLocation
+ */
+function fetchCurrencyPreferences(withLocation) {
+    let coords = {};
+    let currency = '';
+
+    if (withLocation) {
+        Geolocation.getCurrentPosition((position) => {
+            coords = {
+                longitude: position.coords.longitude,
+                latitude: position.coords.latitude,
+            };
+        });
+    }
+
+    API.GetPreferredCurrency({...coords})
+        .then((data) => {
+            currency = data.currency;
+        })
+        .then(API.GetCurrencyList)
+        .then((data) => {
+            const currencyList = JSON.parse(data.currencyList);
+            Onyx.merge(ONYXKEYS.CURRENCY_LIST, currencyList);
+            return currencyList;
+        })
```

Comment:
Isn't this what `getCurrencyList` does? Why do we repeat it instead of calling it directly?